text/2021-02-26-chaos-engg-as-service.md

```diff
@@ -0,0 +1,79 @@
+# Chaos Engineering as a Service (CAAS)
+
+## Summary
+
+Build a unified dashboard to manage and monitor experiments for multiple
+platforms and multiple clusters.
+
+Related Issue: https://github.com/chaos-mesh/chaos-mesh/issues/1462
+
+## Motivation
+
+Following problem exists with the current chaos-mesh design:
+
+1. Cost of maintainance and change is high, since chaos-daemon and chaosd are
+   two seperate programs that serve similar purpose.
+   **Goal is to unify the two.**
+2. Poor observability of experiment results from within the dashboard
+   **Goal is to collect the metrics by Prometheus and show in dashboard.**
```

Comment:
maybe `collect the metrics by Prometheus and show in dashboard` is just one way of improve the observability. 😃

Reply:
@WangXiangUSTC What do you suggest we could do besides Prometheus integration for this?

Reply:
maybe some data of the chaos experiment itself. In fact I don’t have  definite idea yet

Reply:
Good examples of metrics from [Litmus](https://github.com/litmuschaos/chaos-exporter#example-metrics) as mentioned in this [blog](https://chaos-mesh.org/blog/chaos-mesh-remake-one-step-closer-towards-chaos-as-a-service).

I think we must have metrics around pass/fail/awaited for all chaos operations which client is going to perform. We already have one pass metric but we certainly can have more metric around this (like per namespace) to make data more visible and easy to consume.

Reply:
Yeah, I agree with you